File: src/test/java/com/kobylynskyi/graphql/codegen/scala/compile-pass/QueryResolver.scala
import EventStatus._

trait QueryResolver {

    /**
     * Version of the application.
     */
    @throws[Exception]
    def version(): String

    /**
     * List of events of a specified category.
     */
    @throws[Exception]
    def eventsByCategoryAndStatus(categoryId: String, status: EventStatus): scala.Seq[Event]

    /**
     * Single event by ID.
     */
    @throws[Exception]
    def eventById(id: String): Event

    /**
     * Events by IDs.
     */
    @throws[Exception]
    def eventsByIds(ids: scala.Seq[String]): scala.Seq[Event]

}

File: src/test/java/com/kobylynskyi/graphql/codegen/kotlin/compile-pass/QueryResolver.kt

interface QueryResolver {

    /**
     * Version of the application.
     */
    @Throws(Exception::class)
    fun version(): String

    /**
     * List of events of a specified category.
     */
    @Throws(Exception::class)
    fun eventsByCategoryAndStatus(categoryId: String, status: EventStatus?): List<Event?>

    /**
     * Single event by ID.
     */
    @Throws(Exception::class)
    fun eventById(id: String): Event

    /**
     * Events by IDs.
     */
    @Throws(Exception::class)
    fun eventsByIds(ids: List<String>): List<Event>

}
